### PR TITLE
feat: check for incompatible flavors

### PR DESCRIPTION
**Pull request by crgisch**

### CODE REVIEW

**Comment by wpjunior** [comment]
LGTM, please don't forget to generate manifests with `make manifests`

**Review by wpjunior — APPROVED** [review]

**Review by morpheu — APPROVED** [review]
LGTM!

File: internal/pkg/rpaas/k8s.go [hunk 1]
```diff
@@ -1289,6 +1289,11 @@ func (m *k8sRpaasManager) validateFlavors(ctx context.Context, instance *v1alpha
 		if flavorObj.Spec.CreationOnly && !isCreation {
 			return &ValidationError{Msg: fmt.Sprintf("flavor %q can used only in the creation of instance", f)}
 		}
+
+		incompatibleFlavor := checkIncompatibleFlavors(flavorObj.Spec.IncompatibleFlavors, flavors)
```
Comment (morpheu) [inline internal/pkg/rpaas/k8s.go:1293]: I think is just cosmestic, but would be nice if `checkIncompatibleFlavors` returns a error instead a string to be more golang way to handle error

**Review by morpheu — APPROVED** [review]
nice :)

**Pull request merged**